assembly/src/errors.rs

```diff
@@ -273,9 +273,10 @@ impl ParsingError {
         }
     }
 
-    pub fn missing_param(token: &Token) -> Self {
+    pub fn missing_param(token: &Token, expected_format: &str) -> Self {
+        let _actual_params: usize = token.num_parts();
         ParsingError {
-            message: format!("malformed instruction '{token}': missing required parameter"),
+            message: format!("malformed instruction '{token}': expected format {expected_format}"),
```

Comment:
small nit: let's place backticks around `expected_format` so that the message is set to the following:
```Rust
format!("malformed instruction '{token}': expected format `{expected_format}`")
```

Reply:
had to do a little more change here, since the result of `format!("malformed instruction '{token}': expected format `{expected_format}`")` would be a String and this was expected to be `&str`.

Let me know if you think this can work.